lib/web_ui/lib/src/engine/canvaskit/path.dart

```diff
@@ -257,6 +257,7 @@ class CkPath implements ui.Path {
 
   @override
   void reset() {
+    _fillType = ui.PathFillType.nonZero;
```

Comment:
I think you want to do `fillType = ui.PathFillType.nonZero` (otherwise it won't send the value over to the WASM side; see line 37).

Reply:
Ah, nvm. You explained it in the issue. `_skPath.reset()` does that already.

Reply:
That was intentional as that's what the WASM side defaults it to (perhaps more future proof to set it as well).

Reply:
If this is implementing ui.Path.reset(), then it should not modify the winding rule. Instead of hard-coding the rule to nonZero, it should reset the WASM side to the current local value.

Reply:
The underlying call to SkPath::reset does this: https://github.com/google/skia/blob/1b89eb742a66257cd780f119c537e833f4554f53/src/core/SkPath.cpp#L163

The stored _fillType needs to be synced with that or further rendering can use the wrong fillType as the CkPath believes it's stored _fillType is still set on the backing path. This is the behavior we're seeing, where paths with evenOdd will start using nonZero after calling reset and cannot be set back to evenOdd without first changing them nonZero. 

Setting it to nonZero simply does the equivalent of what Skia is doing so that stored value is in sync. The alternative, more future proof way to do this would be to query the fill rule from _skPath and sync the value stored in CkPath that way. Unless the main flutter engine expects a ui.path.reset to force the fillType/Rule to some other value.

Reply:
Ignore my previous comment, I was looking at an old version of the sources (I had been investigating an old bug) and what I was reading implied that the fillType was not supposed to be changed on a reset(), but now I see that this is the correct behavior.